src/web_asset_source.rs

```diff
@@ -83,37 +83,45 @@ async fn get<'a>(path: PathBuf) -> Result<Box<Reader<'a>>, AssetReaderError> {
 
 #[cfg(not(target_arch = "wasm32"))]
 async fn get<'a>(path: PathBuf) -> Result<Box<Reader<'a>>, AssetReaderError> {
-    use std::future::Future;
     use std::io;
-    use std::pin::Pin;
-    use std::task::{Context, Poll};
 
-    use isahc::http::StatusCode;
-    use isahc::{get_async, AsyncBody, Error, Response, ResponseFuture};
-
-    struct ContinuousPoll<'a>(ResponseFuture<'a>);
-
-    impl<'a> Future for ContinuousPoll<'a> {
-        type Output = Result<Response<AsyncBody>, Error>;
-
-        fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
-            // Always wake - blocks on single threaded executor.
-            cx.waker().wake_by_ref();
-
-            Pin::new(&mut self.0).poll(cx)
-        }
-    }
+    use bevy::asset::io::VecReader;
+    use surf::StatusCode;
 
-    let response = ContinuousPoll(get_async(path.to_str().unwrap()))
-        .await
-        .unwrap();
+    let mut response = surf::get(path.to_str().ok_or_else(|| {
+        AssetReaderError::Io(io::Error::new(
+            io::ErrorKind::Other,
+            format!("non-utf8 path: {}", path.display()),
+        ))
+    })?)
+    .await
```

Comment:
For me, the web_image example hangs here. The await never finishes

This works on main and on v0.5.0